manifests/init.pp

```diff
@@ -250,6 +251,7 @@
   Optional[Enum['never', 'allow', 'try', 'demand', 'hard']] $ldap_reqcert     = $zabbix::params::ldap_reqcert,
   $zabbix_api_user                                                            = $zabbix::params::server_api_user,
   $zabbix_api_pass                                                            = $zabbix::params::server_api_pass,
+  Optional[Array[Stdlib::Fqdn]] $zabbix_api_access                            = $zabbix::params::server_api_acces,
```

Comment:
Agree with @root-expert that `Stdlib::Host` might be a better choice.  You've also got a typo at the end of the line (a missing `s` in `access`).
```suggestion
Optional[Array[Stdlib::Host],1] $zabbix_api_access                          = $zabbix::params::server_api_access,
```